src/utils/mod.rs

```diff
@@ -32,42 +36,43 @@ impl Multipolygon {
 // Users of this code must verify correctness for their application.
 // http://geomalgorithms.com/a03-_inclusion.html
 
-fn is_point_in_polygon(p: &Point, v: &[Point]) -> bool {
+fn is_point_in_polygon(p: Point, v: &[Point]) -> bool {
```

Comment:
(Why) is it better if the Point is consumed / automatically copied instead of the same instance be passed around?

Reply:
Ah I see, `clippy` tells me:

> this argument (4 byte) is passed by reference, but would be more efficient if passed by value (limit: 8 byte)

Reply:
yep, clippy told me too.  In a more general aspect -- passing by ref means you pass around 64bit value which then has to be dereferenced.  Dereferencing is an expensive operation - given an address, CPU has to go to the memory unless cached, etc. The by-value would be passed in a register. So by value is always better for small-ish values... where the size of "small" is at least 64bit, but actually could be higher (passing a 128bit value might still be faster than passing a 64bit value plus memory call) - but that requires [benchmarking](https://bheisler.github.io/criterion.rs/book/getting_started.html)